pkg/common/resources/resources.go

```diff
@@ -816,6 +874,29 @@ func ComponentWiseMinPermissive(left, right *Resource) *Resource {
 	return out
 }
 
+// ComponentWiseMinOnlyExisting Returns a new Resource with the smallest value for resource type
+// existing only in left but not vice versa.
+func ComponentWiseMinOnlyExisting(left, right *Resource) *Resource {
```

Comment:
This is the 3rd version of `ComponentWiseMin...` need to think about a refactor in a follow up jira